Error when calling `appWs.callZome({ role_name: '', ... })`

Latest PR https://github.com/holochain/tryorama/pull/201/files removed the ability to call `appWs.callZome({ role_name, ... })`, using role name instead of cell id to identify the cell. This is the technique I use in all my tryorama tests, which are actually also testing my real UI code, which calls the `AppClient` using the role_name.

This is the error I get:

```
⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯ Failed Tests 1 ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯

 FAIL  src/profile.test.ts > create Profile
TypeError: Cannot read properties of undefined (reading '0')
 ❯ getSigningCredentials ../node_modules/@holochain/client/lib/api/zome-call-signing.js:13:48
 ❯ AppWebsocket.appWs.callZome ../node_modules/@holochain/tryorama/lib/local/conductor.js:250:12
 ❯ ProfilesClient.callZome ../node_modules/@holochain-open-dev/utils/src/zome-client.ts:32:24
 ❯ ProfilesClient.getAgentsWithProfile ../ui/src/profiles-client.ts:53:17
     51|    */
     52|   async getAgentsWithProfile(): Promise<Link[]> {
     53|     return this.callZome("get_agents_with_profile", null);
       |                 ^
     54|   }
     55|
 ❯ ../ui/src/profiles-store.ts:39:25
 ❯ fetch ../node_modules/@holochain-open-dev/signals/src/holochain.ts:92:11
 ❯ e.[watched] ../node_modules/@holochain-open-dev/signals/src/holochain.ts:106:9
 ❯ K ../node_modules/signal-polyfill/dist/index.js:136:34
 ❯ K ../node_modules/signal-polyfill/dist/index.js:138:34
 ❯ L ../node_modules/signal-polyfill/dist/index.js:79:88

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[1/1]⎯

 Test Files  1 failed (1)
      Tests  1 failed (1)
   Start at  19:17:54
   Duration  18.04s (transform 628ms, setup 0ms, collect 2.38s, tests 14.74s, environment 1ms, prepare 354ms)

npm ERR! Lifecycle script `test` failed with error:
npm ERR! Error: command failed
npm ERR!   in workspace: tests
npm ERR!   at location: /home/guillem/projects/holochain/open-dev/profiles/tests
```

It's because in [this line](https://github.com/holochain/tryorama/blob/f6b5b3c6b5eddf4682c44ef0eabc4e600cced2e1/ts/src/local/conductor.ts#L324), there is no cell_id in the request. I think it's because of the removal of [this function](https://github.com/holochain/tryorama/pull/201/files#diff-f4b6f81934e7c5260aa4ccd8897c20f6c2c59f79e2d1ab877184cde197bd3343L356).

Another issue is that I don't see anywhere in the tests for tryorama that `callZome` is tested using the role name instead of the cell_id.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -9,6 +9,10 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 ### Changed
 ### Fixed
 
+## 2024-04-29: v0.16.0-dev.6
+### Fixed
+- Restored the ability to make a zome call on the app websocket using a role name instead of a cell id.
+
 ## 2024-04-27: v0.16.0-dev.5
 ### Changed
 - Update the version of the JS client to v0.17.0-dev.12.

--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
   "name": "@holochain/tryorama",
   "description": "Toolset to manage Holochain conductors and facilitate running test scenarios",
-  "version": "0.16.0-dev.5",
+  "version": "0.16.0-dev.6",
   "author": "Holochain Foundation",
   "license": "MIT",
   "keywords": [

--- a/ts/src/local/conductor.ts
+++ b/ts/src/local/conductor.ts
@@ -9,6 +9,7 @@ import {
   getSigningCredentials,
   InstallAppRequest,
   AppAuthenticationToken,
+  AppCallZomeRequest,
 } from "@holochain/client";
 import getPort, { portNumbers } from "get-port";
 import pick from "lodash/pick.js";
@@ -317,12 +318,19 @@ export class Conductor implements IConductor {
 
     // set up automatic zome call signing
     const callZome = appWs.callZome.bind(appWs);
-    appWs.callZome = async (
-      req: CallZomeRequest | CallZomeRequestSigned,
-      timeout?: number
-    ) => {
-      if (!getSigningCredentials(req.cell_id)) {
-        await this.adminWs().authorizeSigningCredentials(req.cell_id);
+    appWs.callZome = async (req: AppCallZomeRequest, timeout?: number) => {
+      let cellId;
+      if ("role_name" in req) {
+        assert(appWs.cachedAppInfo);
+        cellId = appWs.getCellIdFromRoleName(
+          req.role_name,
+          appWs.cachedAppInfo
+        );
+      } else {
+        cellId = req.cell_id;
+      }
+      if (!getSigningCredentials(cellId)) {
+        await this.adminWs().authorizeSigningCredentials(cellId);
       }
       return callZome(req, timeout);
     };

--- a/ts/src/local/conductor.ts
+++ b/ts/src/local/conductor.ts
@@ -3,8 +3,6 @@ import {
   AppBundleSource,
   AppWebsocket,
   AttachAppInterfaceRequest,
-  CallZomeRequest,
-  CallZomeRequestSigned,
   encodeHashToBase64,
   getSigningCredentials,
   InstallAppRequest,

--- a/ts/test/local/scenario.ts
+++ b/ts/test/local/scenario.ts
@@ -1,4 +1,5 @@
 import {
+  ActionHash,
   AppBundleSource,
   AppSignal,
   AppSignalCb,
@@ -277,4 +278,21 @@ test("Local Scenario - pauseUntilDhtEqual - Create multiple entries, read the la
   t.equal(readContent, lastCreatedContent);
 
   await scenario.cleanUp();
+});
+
+test("Local Scenario - runScenario - call zome by role name", async (t) => {
+  await runScenario(async (scenario: Scenario) => {
+    const alice = await scenario.addPlayerWithApp({
+      path: FIXTURE_HAPP_URL.pathname,
+    });
+
+    const result = (await alice.appWs.callZome({
+      role_name: "test",
+      zome_name: "coordinator",
+      fn_name: "create",
+      payload: "hello",
+    })) as ActionHash;
+
+    t.ok(result);
+  });
 });

